api/v1/api.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"net/http"
+
+	"github.com/GoogleCloudPlatform/heapster/manager"
+	sinksApi "github.com/GoogleCloudPlatform/heapster/sinks/api"
+	"github.com/emicklei/go-restful"
+)
+
+type Api struct {
+	manager manager.Manager
+}
+
+// Create a new Api to serve from the specified cache.
+func NewApi(m manager.Manager) *Api {
+	return &Api{
+		manager: m,
+	}
+}
+
+// Register the Api on the specified endpoint.
+func (a *Api) Register(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/api/v1/metric-export").
+		Doc("Exports the latest point for all Heapster metrics").
+		Produces(restful.MIME_JSON)
+	ws.Route(ws.GET("").To(a.exportMetrics).
+		Doc("export the latest data point for all metrics").
+		Operation("exportMetrics").
+		Writes([]*Timeseries{}))
+	container.Add(ws)
+}
+
+func (a *Api) exportMetrics(request *restful.Request, response *restful.Response) {
+	points, err := a.manager.ExportMetrics()
+	if err != nil {
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	// Group points by labels.
```

Comment:
Does this only group by the target labels (the ones that identify a container), or will it also group by the metric-specific labels, like device_name and fault_type?